### ISSUE TITLE
Radarr unable to postprocess on rclone mount

### ISSUE THREAD

**Issue by ErAzOr2k** [ncw/rclone#2079]
> What is your rclone version (eg output from `rclone -V`)
rclone v1.39-155-g62e72801β
- os/arch: linux/amd64
- go version: go1.9.3

> Which OS you are using and how many bits (eg Windows 7, 64 bit)
Linux (Debian)

> Which cloud storage system are you using? (eg Google Drive)
Google Drive

> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
Postprocessing of Radarr

> A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
see link at the end of the description

Hi,

at the end of each post processing in Radarr, the movie will be moved to the final destination. In my case, it's the rclone mount.

While moving the file, a temporarily file named movie.mkv.partial is written. But after some time, the file suddenly just disappears. A time later a new partial file gets written.

The steps above are repeated over and over again. So it's like an infinite loop.

see logs on pastebin: [https://pastebin.com/duCEjerf](https://pastebin.com/duCEjerf)

**Comment by ErAzOr2k** [comment]
The expected behavior is:
Radarr moves file to the final destination. While moving, name file to movie.mkv.partial.
After move is completed, rename movie.mkv.partial to movie.mkv

**Comment by B4dM4n** [comment]
Did you experience this behavior also in an earlier version of rclone? rclone `v1.39-155-g62e72801β` uses a very recent change in the handling of cached writes. 

From the log you provided it looks like this caused by Radarr.

When Radarr is closing the file for the first time, rclone starts uploading the cached file. Radarr should wait for the file to be closed completely, but instead tries to find the file (`Lookup: name="movie.mkv.partial~"`) and then starts writing it again. Until the end of the log rclone is still uploading the first cache file (`Sending chunk 2726297600 length 8388608`).

I will extend the checks for exclusive writing (`O_CREATE|O_EXCL`) to disallow multiple writes, but this will probably make Radar fail because of the returned error.

**Comment by ErAzOr2k** [comment]
Hi,
thank you for your feedback.
I can confirm, that I experienced this same issue before your pull. So it should have nothing to do with your recent changes.

You say: rclone will only add the file to the directory structure when it has completed the upload.
In fact (at least for me), the file (movie.mkv.partial) is visible on the mount while it is written to the directory.

**Comment by ErAzOr2k** [comment]
Just one thing I forgot to mention. But I think it's a important information:

Radarr is not directly accessing the rclone mount. Instead Radarr is accessing the mount over cifs.
Is it possible, that this issue not caused by Radarr, but caused by cifs?

**Comment by daniel-loader** [comment]
Any reason for exposing the mount over CIFS?

**Comment by ErAzOr2k** [comment]
Well, the only reason is, that love to have everything centralized.

**Comment by daniel-loader** [comment]
I'd start with trying to mount rclone on the same installation as Radarr and try again. Not experiencing the same issue with the latest beta with my Radarr/Rclone mount.

**Comment by Vardex** [comment]
I'm having the same issue.

rclone v1.39-170-g815a40a4β
- os/arch: linux/amd64
- go version: go1.10

Ubuntu 16.04

Google Drive

First 1000 lines of log: https://pastebin.com/fQ1NhuN6

The file disseapered after ~5 minutes but it's still outputing the `DEBUG : &{Movie (2017)/Movie (2017) - Remux-2160p.mkv.partial~ (w)}: Write: len=8192, offset=XXXXXXXX` in the log

**Comment by ErAzOr2k** [comment]
New insights with latest beta (v1.39-184-gb91bd324-vfs-updateβ):
[https://pastebin.com/GC5jwzeN](https://pastebin.com/GC5jwzeN)

With latest beta, I don't have an "infinite loop" like before, but still have the problem.

I think the most interesting part in the log is: 
`2018/02/27 10:17:35 DEBUG : path/to/themovie/: Rename: oldName="movie.mkv.partial~", newName="movie.mkv", newDir=path/to/themovie/
2018/02/27 10:17:35 ERROR : path/to/themovie/movie.mkv.partial~: Dir.Rename cant rename open file
2018/02/27 10:17:35 DEBUG : path/to/themovie/: >Rename: err=operation not permitted`

**Comment by ErAzOr2k** [comment]
Fyi: I don't think it's an radarr issue, as I have same same issue with nextcloud.

Background:
I try to write a file over nextcloud interface directly to the rclone mount. While uploading, the file is written with .part extension. After complete upload, it should be renames to the final filename.
But I'm getting this error (in nextcloud):

Error | webdav | renaming part file to final file failed ($run: true, $renameOkay: false, $fileExists: false)
-- | -- | --

**Comment by ncw** [comment]
@ErAzOr2k wrote:
> Dir.Rename cant rename open file

Hmm, that is an error from the VFS layer...  And it is a bit misleading too!

I would expect this to work with [the latest beta](https://beta.rclone.org) as I put lots of VFS fixes in quite recently and with `--vfs-cache-mode writes`.

Can you try that?

I suspect it won't work though which means that the rename code needs a bit of fixing.

**Comment by ErAzOr2k** [comment]
@ncw thanks for your feedback.
I tried it with the latest beta. Actually the issue is still the same :(

**Comment by ncw** [comment]
That error `Dir.Rename cant rename open file` is either caused by
  1. Radarr renaming an open file
  2. Some bug in rclone

You could make absolutely certain if you strace Radarr and check, but I'm going to assume it is 1. since we've seen that from geany in #2130 

Under unix you are quite entitled to rename open files.  I don't think that works under Windows though.

It would be a reasonable amount of work to get rclone to support renaming open files and that would only work on `--vfs-cache-mode writes` or above.

Worth implementing?  What do you think @B4dM4n ?

**Comment by ErAzOr2k** [comment]
It seems B4dM4n is not active anymore :(
Hope someone else is giving this issue some love

**Comment by breunigs** [comment]
Yup. Since you're on Linux, you can already give it a spin. It fixes saving for Geany, but Radarr looks too complicated to setup. I'll have to see about that Windows failure before merge, though, and Windows isn't exactly nice to work on :(

**Comment by rgdevx** [comment]
Hi! I noticed the same issue when saving files with [medit](http://mooedit.sourceforge.net/) , to a Google Drive folder mounted with rclone.

I get these 2 errors from rclone:
```
2018/04/20 23:38:30 ERROR : file.txt: WriteFileHandle: Can't open for write without O_TRUNC on existing file without --vfs-cache-mode >= writes
2018/04/20 23:38:31 ERROR : .goutputstream-TU9JHZ: Dir.Rename cant rename open file
```

If I enable **--vfs-cache-mode writes** , the first error message disappears but the second one remains ( `Dir.Rename cant rename open file` )

In **medit** I get this error: `Could not save file file.txt Error renaming temporary file: Operation not permitted`

I installed rclone from source, with ` go get -u -v github.com/ncw/rclone ` a few hours ago. I'm testing it on Arch Linux.
```
$ rclone --version
rclone v1.40-DEV
- os/arch: linux/amd64
- go version: go1.10.1

```
